mayastor/src/target/mod.rs

```diff
@@ -6,3 +8,29 @@ pub enum Side {
     Nexus,
     Replica,
 }
+
+/// Types of remote access storage protocols and IDs for sharing replicas.
+pub enum ShareType {
+    Nvmf,
+    Iscsi,
+}
+
+/// Detect share protocol (if any)
+pub fn detect_share(
```

Comment:
If a bdev is shared or not is (as mentioned in the previous comment somewhere, sorry for repeating) is already possible with the Share trait you can see it implemented in bdev.rs